iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/ImportInformationSpaceReducer.java

```diff
@@ -0,0 +1,368 @@
+
+    private static final String OUTPUT_NAME_PERSON = "output.name.person";
+
+    private static final String OUTPUT_NAME_DEDUP_MAPPING = "output.name.dedup_mapping";
+
+    private static final String OUTPUT_NAME_ORGANIZATION = "output.name.organization";
+
+    private static final String OUTPUT_NAME_PROJECT_ORGANIZATION = "output.name.project_organization";
+
+    // column family names
+    
+    private final String projOrgColumnFamily = OafRelDecoder.getCFQ(RelType.projectOrganization,
+            SubRelType.participation, ProjectOrganization.Participation.RelName.hasParticipant.toString());
+    
+    private final String resProjColumnFamily = OafRelDecoder.getCFQ(RelType.resultProject, 
+            SubRelType.outcome, Outcome.RelName.isProducedBy.toString());
+    
+    private final String dedupMappingColumnFamily = OafRelDecoder.getCFQ(RelType.resultResult, 
+            SubRelType.dedup, Dedup.RelName.merges.toString());
+
+    // output names
+    
+    private String outputNameDocumentMeta;
+
+    private String outputNameDocumentProject;
+
+    private String outputNameProject;
+
+    private String outputNamePerson;
+
+    private String outputNameDedupMapping;
+
+    private String outputNameOrganization;
+
+    private String outputNameProjectOrganization;
+
+    // converters
+    
+    private DocumentMetadataConverter docMetaConverter;
+
+    private DocumentToProjectConverter docProjectConverter;
+
+    private DeduplicationMappingConverter deduplicationMappingConverter;
+
+    private PersonConverter personConverter;
+
+    private ProjectConverter projectConverter;
+
+    private OrganizationConverter organizationConverter;
+
+    private ProjectToOrganizationConverter projectOrganizationConverter;
+
+    // others
+    
+    private String encoding = HBaseConstants.STATIC_FIELDS_ENCODING_UTF8;
+    
+    private MultipleOutputs mos;
+    
+    private ResultApprover resultApprover;
+    
+    /**
+     * Flag indicating {@link Oaf} retrieved from body column family should be merged with all update collumns. 
+     * Set to false by default.
+     */
+    private boolean mergeBodyWithUpdates;
+    
+    // ------------------------ LOGIC --------------------------
+    
+    @Override
+    public void setup(Context context) {
+        setOutputDirs(context);
+
+        if (context.getConfiguration().get(HBASE_ENCODING) != null) {
+            encoding = context.getConfiguration().get(HBASE_ENCODING);
+        }
+        mergeBodyWithUpdates = context.getConfiguration().get(IMPORT_MERGE_BODY_WITH_UPDATES) != null
+                ? Boolean.valueOf(context.getConfiguration().get(IMPORT_MERGE_BODY_WITH_UPDATES)) : false;
+
+        DataInfoBasedApprover dataInfoBasedApprover = buildApprover(context);
+        this.resultApprover = dataInfoBasedApprover;
+
+        // initializing converters
+        docMetaConverter = new DocumentMetadataConverter(this.resultApprover, dataInfoBasedApprover);
```

Comment:
Using `resultApprover` indicates we are dealing with `ResultApprover` impl while `dataInfoBasedApprover` implements both `ResultApprover` and `FieldApprover` and this is why this reference is passed as `fieldApprover`. 

I could pass the same `dataInfoBasedApprover` reference for both.